backend/server.js

```diff
@@ -186,12 +186,84 @@ app.get("/user-activities", authenticateToken, async (req, res) => {
   try {
     const activities = await prisma.userActivity.findMany({
       where: { userId: req.user.id },
-      orderBy: { startDateTime: 'desc' },
-      take: 10 // Limit to 10 most recent activities
+      orderBy: { startDateTime: "desc" },
+      take: 10, // Limit to 10 most recent activities
     });
     res.json(activities);
   } catch (error) {
-    res.status(500).json({ message: "Error fetching activities", error: error.message });
+    res
+      .status(500)
+      .json({ message: "Error fetching activities", error: error.message });
+  }
+});
+
+// Function to calculate calories burned based on user profile and route
+
+const calculateCaloriesBurned = (user, distance, elevationGain) => {
+  // This is a simplified calculation and should be refined for more accuracy
+  const weight = user.weight * 0.453592; // Convert lbs to kg
+  const duration = parseFloat(distance) * 10; // Assume 10 minutes per mile
+  const caloriesPerMinute = 0.0175 * 8 * weight; // MET value of 8 for running
```

Comment:
nit: you could add some of these values as constants rather than having them as magic numbers in the code. Up to you tho, it's not a huge deal since you explain what's going on in your comments

Reply:
agreed! you have been adding more mathematical calculations lately and you could consider creating a constants file for defining all of these hard-coded values you have